File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.py
from collections import deque
def solution(bridge_length, weight, truck_weights):
    time= 0
    
    road = deque([0]*bridge_length)
    truck = deque(truck_weights)
    cur = 0
    while truck:
        time+=1
        
        cur -= road.popleft()
        
        if cur + truck[0] <= weight:
            cur += truck[0]
            road.append(truck.popleft())
        else:
            road.append(0)
        
        
    if len(road):
        time += len(road)
        

    
    return time

File: 프로그래머스/2/42583. 다리를 지나는 트럭/다리를 지나는 트럭.java
import java.util.*;
class Solution {
    public int solution(int bridge_length, int weight, int[] truck_weights) {
        int answer = 0;
        Queue<Integer> bridge = new LinkedList<>();
        int n= truck_weights.length;
        for(int i=0;i<bridge_length;i++){
            bridge.offer(0);
        }
        int l = 0;
        int time = 0;
        int totalWeight = 0;
        while(l<n){
            time++;
            totalWeight -= bridge.poll();
            if(totalWeight + truck_weights[l] <= weight ){
                bridge.offer(truck_weights[l]);
                totalWeight += truck_weights[l];
                l++;
            }else{
                bridge.offer(0);
            }
        }
        
        return time+bridge_length;
    }
}